### ISSUE TITLE
False positive with custom `int` type.

### ISSUE THREAD

**Issue by beeb** [python/mypy#10629]
**Bug Report**

When defining a new type based on `int`, mypy will complain about incompatible types in assignment if performing operations on a variable of said type.

**To Reproduce**

```python
from typing import NewType

Wei = NewType('Wei', int)

a = Wei(1)

a += Wei(2)  # incompatible types in assignment
```

**Expected Behavior**

There should be no error here.

**Actual Behavior**

A mypy error is raised.

**Your Environment**

- Mypy version used: `0.902`
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: `3.7.10`
- Operating system and version: WSL2 Ubuntu 20.04

As far as I remember, the error was not present in `mypy 0.812` and started appearing after I upgraded my dependencies. Can't be sure of when this changed.

**Comment by beeb** [comment]
Actually this seems to be the expected behavior: https://docs.python.org/3/library/typing.html#newtype
